### ISSUE TITLE
Replicas are not re-balanced (back) when using RSP and "rebalance: true"

### ISSUE THREAD

**Issue by mvazquezc** [kubernetes-sigs/kubefed#763]
**What happened**:
A RSP[1] is created to distribute replicas evenly across all clusters (3 in our scenario), one of the clusters is cordoned so replicas cannot be scheduled there. The rebalancing mechanism configured the replicas like this:

```
Cluster A: 5
Cluster B: 0
Cluster C: 4
```
If we uncordon Cluster B, the replicas are not re-balanced, so Cluster B configuration remains to 0 replicas.

[1]
```
apiVersion: scheduling.federation.k8s.io/v1alpha1
kind: ReplicaSchedulingPreference
metadata:
  name: reverse-words
spec:
  targetKind: FederatedDeployment
  totalReplicas: 9
  clusters:
    "*":
      weight: 1
```

**What you expected to happen**:

I would expect the replicas to be re-balanced like this:

```
Cluster A: 3
Cluster B: 3
Cluster C: 3
```

**How to reproduce it (as minimally and precisely as possible)**:
Create the RSP[1] targeting a FederatedDeployment with `rebalance: true` configured. Cordon worker nodes in one of the target clusters. Wait until the replicas are rebalanced across the available clusters. 

Once the replicas have been rebalanced, uncordon the cordoned nodes and check out that replicas are not rebalanced.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`)
```
Client Version: version.Info{Major:"4", Minor:"1+", GitVersion:"v4.1.0", GitCommit:"1016bda6b", GitTreeState:"", BuildDate:"2019-04-09T04:48:59Z", GoVersion:"", Compiler:"", Platform:""}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.4+0ba401e", GitCommit:"0ba401e", GitTreeState:"clean", BuildDate:"2019-03-31T22:28:12Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```
- Federation version
```
v0.0.7
```
- Install method (helm vs script)
```
Operator Marketplace
```
- Install configuration (namespaced vs cluster-scoped)
```
Namespaced
```
- Others

/kind bug

**Comment by marun** [comment]
@irfanurrehman Is this expected behavior?

**Comment by marun** [comment]
This should be fixed by #1087.  Feel free to re-open if this is not the case.
